shakytweaks/src/main/java/com/mindsea/shakytweaks/ShakyTweaks.kt

```diff
@@ -24,31 +24,37 @@
 
 package com.mindsea.shakytweaks
 
+import android.app.Activity
 import android.content.Context
 import android.hardware.Sensor
 import android.hardware.SensorManager
 import android.view.KeyEvent
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.LifecycleOwner
 import com.mindsea.shakytweaks.ui.createTweaksActivityIntent
 import com.mindsea.shakytweaks.util.ShakeDetector
 import java.util.*
 
 private const val SHARED_PREFERENCES_NAME = "shaky_tweaks"
 
-object ShakyTweaks {
+object ShakyTweaks: LifecycleEventObserver {
 
-    private var isInitialized: Boolean = false
     private val moduleImpl = LibraryModuleImpl()
 
+    private val shakeDetector = ShakeDetector()
+
     private var sKeyPressMS = 0L
     private var tKeyPressMS = 0L
 
     private const val KEY_PRESS_THRESHOLD = 50L
 
+    private lateinit var sensorManager: SensorManager
+    private lateinit var accelerometer: Sensor
+    private lateinit var lifecycle: Lifecycle
+
+    @Deprecated("Issues related to the lifecycle.\n", replaceWith = ReplaceWith("init(activity: Activity, lifecycle: Lifecycle)"))
```

Comment:
If we are changing a a major number, because of breaking changes, I think this method could be removed

Reply:
Okay I will remove so ;)

Reply:
Done!